src/termdict/fst_termdict/streamer.rs

```diff
@@ -149,3 +149,153 @@ where
         }
     }
 }
+
+/// `TermWithStateStreamerBuilder` is a helper object used to define
+/// a range of terms that should be streamed.
+pub struct TermWithStateStreamerBuilder<'a, A = AlwaysMatch>
```

Comment:
I think it is ok to just have a TermWithStateStreamer and avoid the code duplication.

Most of the time the state is small and "Copy".

Reply:
So is it OK if I remove the extra `TermWithState*` structs I added, and instead add the state to all the `TermStreamer*` structs as @maufl also suggested ?

Reply:
Yes